File: Leetcode/3136 - Valid Word/solution.py
def isValid(self, word: str) -> bool:
    if len(word) < 3:
        return False

    vowels: Set[str] = {"a", "e", "i", "o", "u"}
    has_vowel = False
    has_consonant = False

    for ch in word:
        if ch.isalpha():
            ch_lower = ch.lower()
            if ch_lower in vowels:
                has_vowel = True
            else:
                has_consonant = True
        elif not ch.isdigit():
            return False

    return has_vowel and has_consonant

File: Leetcode/3136 - Valid Word/solution.cpp
bool isValid(string word)
{
    if (word.size() < 3)
        return false;

    unordered_set<char> vowels = {'a', 'e', 'i', 'o', 'u'};
    bool hasVowel = false, hasConsonant = false;

    for (char ch : word)
    {
        if (isalpha(ch))
        {
            ch = tolower(ch);
            if (vowels.count(ch))
            {
                hasVowel = true;
            }
            else
            {
                hasConsonant = true;
            }
        }
        else if (!isdigit(ch))
        {
            return false;
        }
    }

    return hasVowel && hasConsonant;
}